pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -371,6 +369,17 @@ cdef inline int64_t parse_timedelta_string(str ts) except? -1:
             have_value = 1
             have_dot = 0
 
+    # Consider units from outside
```

Comment:
why are we doing the checking at this low level? this is simply a check on the user interface, e.g. if its a string, then no unit can be specified.

see how we do this in to_datetime